### PR TITLE
Add Black Formatting

### PR DESCRIPTION
**Pull request by EthanHindmarsh**
# Changes
- Add Black Formatting workflows (Taken from https://github.com/cclauss/autoblack) 👍
- - Format all code with the Black formatter ( See [Black's Documentation](https://black.readthedocs.io/en/stable/) or [Black's PyPi page](https://pypi.org/project/black/))
- Update CONTRIBUTING.md 

This change is entirely a style change -- There will be no impact on the code's performance from this
**requirements.txt does not need to be updated** -- this is ***not*** a requirement for running the code

# Fixes 
- Inconsistent use of ' and "
- Inconsistent use of tabs and spaces (if any exist)
- (Tries to) Ensures code readability
- Makes code style uniform
- Update all code to follow [PEP 8](https://www.python.org/dev/peps/pep-0008/) (Read about Black's style [here](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html))

## Checklist:

- [x] My code follows the style guidelines of this project and have read CONTRIBUTING.md (**This adds style guidelines**, I could not find any for this project)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works (I plan to add tests later)
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by rachmadaniHaryono** [comment]
i don't know if this pr will be accepted, but maybe add pre commit black or use darker https://github.com/akaihola/darker

**Comment by EthanHindmarsh** [comment]
Yes, pre-commit black would be the best option👍
Not every PR/push is made on the desktop, which is why I included the
workflow.

I'm 100% sure some commits would slip by unformatted if the workflow was
removed,

The Autoblack workflow has a really neat feature to automatically edit PR's
or create PR's if any push does not pass Black's check

I'd personally set it to do a daily run over the repo and run on every new
PR, but that's just my opinion, so I left it as it was set in Autoblack's
source

**Comment by EthanHindmarsh** [comment]
If Theo wishes for the repo to have style guidelines, it'll be best to get
them done sooner rather than later -- Merging already existing PRs after a
large restyling like this could be a bit annoying, best to do it when there
are few open PRs👍
If not, I'd get it removed from the pre-PR checklist 😂

**Comment by rachmadaniHaryono** [comment]
> If Theo wishes for the repo to have style guidelines, it'll be best to get
them done sooner rather than later -- Merging already existing PRs after a
large restyling like this could be a bit annoying, best to do it when there
are few open PRs

i think only #424 change the main program

 other than that other pr mainly for non python file

**Comment by ThioJoe** [comment]
I have literally no idea what this means

**Comment by EthanHindmarsh** [comment]
It's about consistency across the codebase --
Some decent resources on the topic are:
https://searchbusinessanalytics.techtarget.com/tip/Python-code-formatting-Tools-you-need-and-why-it-matters
https://youtu.be/j1MbEYhYj_Y
And
https://www.python.org/dev/peps/pep-0008/

A great quote from the Python's PEP 8 is
"One of Guido's key insights is that code is read much more often than it
is written"
Best to keep the codebase readable👍

**Comment by rachmadaniHaryono** [comment]
simpler 

- add github action to reformat pr
- update contributing.md 

is it too much @ThioJoe? if yes, maybe reduce the pr to essential and let thiojoe reformat the codebase

**Comment by EthanHindmarsh** [comment]
The majority of the commit is changing single quotes to double quotes (and
likely tabs to spaces) to keep the code consistent, but at a quick glance
of the changes, there are definitely some big improvements

As a note to This: none of this changes any of the code, only changes the
way it looks


```diff
[...]
-            text =
post['contentText']['runs'][0]['text'].strip().replace('\n',
'').replace('\r', '')
+            text = (
+                post["contentText"]["runs"][0]["text"]
+                .strip()
+                .replace("\n", "")
+                .replace("\r", "")
+            )
[...]
```
(Deleted original to fix markdown)

File: .github/workflows/black-formatting.yml
Comment (EthanHindmarsh) [inline .github/workflows/black-formatting.yml:64]: In .github/workflows/black-formatting.yml, 
If you have already setup a Github token secret, you can uncomment these lines to allow the workflow to automatically format the code on every pull request or push
```diff
diff --git a/.github/workflows/black-formatting.yml b/.github/workflows/black-formatting.yml
new file mode 100644
index 0000000..6d1ccfe
--- /dev/null
+++ b/.github/workflows/black-formatting.yml
@@ -0,0 +1,64 @@
+# Taken from https://github.com/cclauss/autoblack
+#-- Removed the functionality to automatically fix (can be re-added)
[...]
+# See https://github.com/cclauss/autoblack/blob/master/.github/workflows/autoblack_pull_request.yml
+# See also https://github.com/cclauss/autoblack/blob/master/.github/workflows/autoblack_push.yml
[...]
+name: autoblack-push
+on:
+  push:
+    branches: [ main ]
[...]
+#      - name: If needed, commit black changes to a new pull request
+#        if: failure()
+#        run: |
+#          black .
+#          git config --global user.name github-actions
+#          git config --global user.email '${GITHUB_ACTOR}@users.noreply.github.com'
+#          # git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/$GITHUB_REPOSITORY
+#          git commit -am "fixup! Format Python code with psf/black push"
+#          git push  # --force origin HEAD:$GITHUB_REF
+
+name: autoblack-pull
+on:
+  pull_request:
+    branches: [ main ]
[...]
+#      - name: If needed, commit black changes to the pull request
+#        if: failure()
+#        run: |
+#          printenv | grep GITHUB
+#          git config --global user.name 'cclauss'
+#          git config --global user.email 'cclauss@users.noreply.github.com'
+#          git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/$GITHUB_REPOSITORY
+#          git remote -v
+#          git branch
+#          git status
+#          black .
+#          git status
+#          echo ready to commit
+#          git commit -am "fixup! Format Python code with psf/black pull_request"
+#          echo ready to push
+#          git push
+#          # git push --force origin
```

**Comment by EthanHindmarsh** [comment]
> i don't know if this pr will be accepted, but maybe add pre commit black or use darker https://github.com/akaihola/darker

@ThioJoe Also, if you are unaware of what a pre-commit is, I'd recommend checking out https://githooks.com/

The .git/hooks/pre-commit file tells git what to run before commiting files, 
To add the black formatter to your pre-commit hooks, adding 
```
python3 -m black
```
to the file should work👍

**Comment by EthanHindmarsh** [comment]
Example run of black-push workflow (with the "auto fix" feature commented out)
https://github.com/EthanHindmarsh/YT-Spammer-Purge/runs/5031727716?check_suite_focus=true

**Comment by dekoza** [comment]
Adding [pre-commit](pre-commit.com/) would also benefit this project.

**Comment by KendallDoesCoding** [comment]
> Yes, pre-commit black would be the best option👍 Not every PR/push is made on the desktop, which is why I included the workflow. I'm 100% sure some commits would slip by unformatted if the workflow was removed, The Autoblack workflow has a really neat feature to automatically edit PR's or create PR's if any push does not pass Black's check I'd personally set it to do a daily run over the repo and run on every new PR, but that's just my opinion, so I left it as it was set in Autoblack's source

Yes, I use the desktop app sometimes, but I mostly use the web application to commit.

**Comment by EthanHindmarsh** [comment]
> let thiojoe reformat the codebase

Working on it right now :+1:

**Comment by UP929312** [comment]
> I have literally no idea what this means

It's formatting which is objective so that everyone has to agree on following the same convention, that's what Black is, a forceful formatter that is pretty standard across python scripts.
A pre-commit hook or a workflow will mean that every time someone tries to add something to the codebase, it'll get automatically blackened (so automatically formatted) and is also pretty standard.

It has no effect on code functionality and is purely "cosmetic" in the sense that it's just so that everything follows formatting conventions.

**Comment by EthanHindmarsh** [comment]
Oops -- Workflow made that `fixup!` commit -- See https://github.com/EthanHindmarsh/YT-Spammer-Purge/actions/runs/1784682010
At least it shows it works 🤣

**Comment by EthanHindmarsh** [comment]
@ThioJoe The [changes](https://github.com/ThioJoe/YT-Spammer-Purge/pull/464/files) should be ready for review -- The workflow ([example run](https://github.com/EthanHindmarsh/YT-Spammer-Purge/runs/5038769759?check_suite_focus=true)) will automatically style & format the codebase after you merge this PR (if you do), so I recommend merging any other PRs you want before merging in this one.
Only four files added:
- Two workflows
- - Run Black formatter on push
- - Run Black formatter on pull requests
- Update CONTRIBUTING.md
- Add `.pre-commit-config.yaml` for those who use the [pre-commit python package](https://pre-commit.com/)

**Comment by EthanHindmarsh** [comment]
Changes reverted in [Revert "Fix formatting issues with long dictionary names"](https://github.com/ThioJoe/YT-Spammer-Purge/pull/464/commits/47f5e99b8eefaf5462e1d5aaa8d2b3ccffa0bae0) can be re-added in a separate pull request should this be accepted 👍

**Review by EthanHindmarsh — COMMENTED** [review]
CONTRIBUTING.md should get a complete update, but I think adding a note about Black should be enough for this PR

**Comment by rachmadaniHaryono** [comment]
if you use pre commit, you may don't need to think about it

because every commit will be autoformatted

maybe some note on how to setup your fork on contributing.md, example below

---

## set up your fork

- fork the repo https://docs.github.com/en/get-started/quickstart/fork-a-repo
- create virtualenv https://packaging.python.org/en/latest/guides/installing-using-pip-and-virtual-environments/#creating-a-virtual-environment
- activate virtual env
- install package `pip3 install -r requirements.txt`
- install pre-commit `pip3 install pre-commit`
- install pre-commit to git hook `python3 -m pre-commit install`

**Comment by EthanHindmarsh** [comment]
> if you use pre commit, you may don't need to think about it
> 
> because every commit will be autoformatted

Not everyone commits from desktop:

> Yes, I use the desktop app sometimes, but I mostly use the web application to commit/make PR's


> maybe some note on how to setup your fork on contributing.md, example below

Yeah for sure :+1:
See #489 -- His modified [CONTRIBUTING.md](https://github.com/ThioJoe/YT-Spammer-Purge/blob/97a07f3a65ac4ed3f7f5574a16123d2bbaf0f86b/CONTRIBUTING.md) is great

**Comment by rachmadaniHaryono** [comment]
> Yeah for sure +1 See #489 -- 
> [His modified CONTRIBUTING.md](https://github.com/ThioJoe/YT-Spammer-Purge/blob/97a07f3a65ac4ed3f7f5574a16123d2bbaf0f86b/CONTRIBUTING.md) is great

i have read it

unfortunately the pr change too much

but it is mostly correct

- it describe how to setup the fork
- how to check documentation
- format/check/test the repo

i hope @dekoza can split  the pr so better contributing.md can be used

**Comment by dekoza** [comment]
@rachmadaniHaryono I did what I could to minimize the changes. I used [copier-poetry](https://github.com/pawamoy/copier-poetry) to create the basic directory structure and then moved most of the files into proper places. Then I refactored imports so that everything should work. Right now - since there are no tests - I'm trying to check if everything still works fine. If you ignore those several files added by `copier-poetry`, you'll see that the changes of imports are in fact small.

**Comment by EthanHindmarsh** [comment]
Fetch upstream to ensure no merge conflicts ^^

**Comment by rachmadaniHaryono** [comment]
dekoza close 489, maybe you can copy pretty-format-yaml hook to make sure all yaml file have the same format

**Comment by EthanHindmarsh** [comment]
> dekoza close 489, maybe you can copy pretty-format-yaml hook to make sure all yaml file have the same format

I updated .pre-commit-config.yaml to match #489 
I'm not sure anything else from #489 is relevant to adding Black formatting -- the purpose of this PR

Maybe pyproject, but that seems more of a poetry thing

**Comment by EthanHindmarsh** [comment]
Also, the failed check can be ignored, obviously, the current state of the repo would not pass a Black formatting check 😁

File: .pre-commit-config.yaml
Comment (rachmadaniHaryono) [inline .pre-commit-config.yaml:16]: imo it should be for later when program have better type hint

File: .pre-commit-config.yaml
Comment (rachmadaniHaryono) [inline .pre-commit-config.yaml:23]: it can't be used because it use duty

File: .pre-commit-config.yaml
Comment (rachmadaniHaryono) [inline .pre-commit-config.yaml:41]: maybe add it later when project have poetry config

File: .pre-commit-config.yaml
Comment (rachmadaniHaryono) [inline .pre-commit-config.yaml:48]: i do want this added but this maybe out of scope from this pr (but maybe added it anyway?)

**Review by rachmadaniHaryono — CHANGES_REQUESTED** [review]
also run pre commit autoupdate to make sure it use latest version

**Comment by rachmadaniHaryono** [comment]
> I'm not sure anything else from https://github.com/ThioJoe/YT-Spammer-Purge/pull/489 is relevant to adding Black formatting -- the purpose of this PR

as far as i can tell only some hook pre commit

**Comment by EthanHindmarsh** [comment]
> also run pre commit autoupdate to make sure it use latest version

Doing this right now

As for the other issues you mentioned, I'll just comment out those parts of the yaml file for later 👍

File: .pre-commit-config.yaml
Comment (EthanHindmarsh) [reply .pre-commit-config.yaml:23]: Not sure what duty is, I'll just comment it out

Comment (rachmadaniHaryono) [reply .pre-commit-config.yaml:23]: https://github.com/pawamoy/duty

but it require duties.py on project

**Review by rachmadaniHaryono — APPROVED** [review]

**Comment by ThioJoe** [comment]
Tbh I don't really have any intention of adding this, I don't really understand the benefit

**Comment by dekoza** [comment]
> Tbh I don't really have any intention of adding this, I don't really understand the benefit

Another brick in the wall...

**Comment by Masamune3210** [comment]
Dude chill out, its not your repo. If you don't like a choice you don't have to deal with it lol

**Comment by KendallDoesCoding** [comment]
> Dude chill out, its not your repo. If you don't like a choice you don't have to deal with it lol

> Comment by [Masamune3210](https://github.com/Masamune3210)

> This comment was directed to someone else, not towards the PR or myself
@EthanHindmarsh 

Who was this comment directed to @Masamune3210

**Comment by EthanHindmarsh** [comment]
> > Dude chill out, its not your repo. If you don't like a choice you don't have to deal with it lol
> 
> > Comment by [Masamune3210](https://github.com/Masamune3210)
> 
> > This comment was directed to someone else, not towards the PR or myself
> > @EthanHindmarsh
> 
> Who was this comment directed to @Masamune3210

image
Probably the comment above😂

**Comment by KendallDoesCoding** [comment]
> > > Dude chill out, its not your repo. If you don't like a choice you don't have to deal with it lol
> > 
> > 
> > > Comment by [Masamune3210](https://github.com/Masamune3210)
> > 
> > 
> > > This comment was directed to someone else, not towards the PR or myself
> > > @EthanHindmarsh
> > 
> > 
> > Who was this comment directed to @Masamune3210
> 
> image Probably the comment above😂

Ok lol, I think he was just joking though.

**Pull request closed without merging**
